awsx/ec2/subnetDistributor.ts

```diff
@@ -201,6 +235,55 @@ function cidrSubnetV4(ipRange: string, newBits: number, netNum: number): string
   return `${newAddress}/${newSubnetMask}`;
 }
 
+export function validateRanges(specs: SubnetSpec[]) {
+  const ranges = specs
+    .map((spec) => new ipAddress.Address4(spec.cidrBlock))
+    .map((addr) => {
+      const start = addr.startAddress().bigInteger();
+      const end = addr.endAddress().bigInteger();
+      return { addr, start, end };
+    });
+  let previous = ranges[0];
+  for (const current of ranges.slice(1)) {
```

Comment:
Is it possible for addresses to overlap one apart? ex. index 1 with index 3

Reply:
All the allocations are done incrementally so we're just finding the next based on the previous so I think we're okay. Though one edge case is if the user specified masks significantly exceed the space available within the first AZ, causing it to massively overlap into the next AZ. I don't think we need to worry about that for our tests yet though.